packages/esm-consultation-workflow/src/form/obs.component.tsx

```diff
@@ -92,15 +100,22 @@ const CheckboxObs = <T,>(props: ObsProps<T>) => {
     return <ErrorState headerTitle={t('errorLoadingConcept', { uuid: props.conceptUuid })} error={error} />;
   }
 
-  const selectedItems = concept.answers.filter((conceptAnswer) =>
-    (field.value as string[]).includes(conceptAnswer.uuid),
+  const filteredAnswers = concept.answers.filter((c) => {
+    if (props.filterConceptUuid) {
+      return props.filterConceptUuid.includes(c.uuid);
+    }
+    return true;
+  });
```

Comment:
Desta forma evitamos ter sempre que percorrer o Array de answers.
```suggestion
    const filteredAnswers = props.filterConceptUuid?.length
      ? concept.answers.filter((c) => props.filterConceptUuid.includes(c.uuid))
      : concept.answers;
```

Reply:
Já fiz os testes e implementei a correcção do método `filteredAnswers`